src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -947,6 +1005,122 @@ public static function create_lock( $lock_name, $release_timeout = null ) {
 	public static function release_lock( $lock_name ) {
 		return delete_option( $lock_name . '.lock' );
 	}
+	/**
+	 * Moves the plugin/theme being updated into a temp-backup directory.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+	 *
+	 * @param array $args Array of data for the temp-backup. Must include a slug, the source, and directory.
+	 * @return bool|WP_Error
+	 */
+	public function move_to_temp_backup_dir( $args ) {
+		global $wp_filesystem;
+
+		if ( empty( $args['slug'] ) || empty( $args['src'] ) || empty( $args['dir'] ) ) {
+			return false;
+		}
+
+		/**
```

Comment:
```suggestion
		/*
```

Multiline comments have a single asterisks, docblocks have two.